domain/machine/state/state.go

```diff
@@ -857,3 +858,104 @@ WHERE  name = $machineName.name`
 		return nil
 	})
 }
+
+// LXDProfiles returns the names of the LXD profiles on the machine.
+func (st *State) LXDProfiles(ctx context.Context, mUUID string) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	machineUUIDQuery := machineUUID{UUID: mUUID}
+	query := `
+SELECT &lxdProfile.name
+FROM   machine_lxd_profile
+WHERE  machine_uuid = $machineUUID.uuid`
```

Comment:
Ensure you order by index on the way out.

Reply:
Done, see 82a3a302d8.